trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/SigningContext.java

```diff
@@ -15,5 +15,9 @@
 
 public interface SigningContext
 {
-    String authorization();
+    ParsedAuthorization signingAuthorization();
+
+    byte[] signingKey();
```

Comment:
I'm concerned about exposing this which is why I had the downcasting and only exposed it in `InternalSigningContext`. I'd like to keep that if you don't mind.

Reply:
I don't mind keeping it private, but out of curiosity - what is the concern with this field? The SPI exposes the `Credentials` pair, which contains the access/secret keys - those are more powerful than the per-request signing key

Reply:
It's more of a gut reaction. This is an internal value used only in the AWS-specific signer. I don't like leaking that into the general SPI

Reply:
Fair point. The SHA256 hasher being initialised in ChunkSigner could also be a leaky abstraction, but we could leave that for now.

What do you think about putting the `ChunkSigner` in the `SigningContext`, instead of putting the signing key there?

That way `SigningContext` becomes the place where we put all that's needed to process a request we are dealing with

Reply:
That might work. Give it a try.